### PR TITLE
Feature/result profiles with pydantic

### PR DESCRIPTION
**Pull request by mike0sv**

### CODE REVIEW

File: src/evidently/metrics/data_drift/target_by_features_table.py [hunk 1]
```diff
@@ -43,17 +44,17 @@ class Config:
...
 class TargetByFeaturesTable(Metric[TargetByFeaturesTableResults]):
     columns: Optional[List[str]]
-    text_features_gen: Optional[
+    _text_features_gen: Optional[
         Dict[
             str,
             Dict[str, Union[TextLength, NonLetterCharacterPercentage, OOVWordsPercentage]],
         ]
-    ]
+    ] = PrivateAttr(None)
```
Comment (Liraim) [inline src/evidently/metrics/data_drift/target_by_features_table.py:52]: Is `PrivateAttr(None)` needed here?

File: src/evidently/metrics/data_integrity/column_regexp_metric.py [hunk 2]
```diff
@@ -58,13 +59,14 @@ class ColumnRegExpMetric(Metric[DataIntegrityValueByRegexpMetricResult]):
     reg_exp: str
     top: int
     # compiled regular expression for speed optimization
-    _reg_exp_compiled: Pattern
+    _reg_exp_compiled: Pattern = PrivateAttr()
```
Comment (Liraim) [inline src/evidently/metrics/data_integrity/column_regexp_metric.py:62]: Is `PrivateAttr` needed here also?

File: src/evidently/base_metric.py [hunk 3]
```diff
@@ -216,4 +243,12 @@ def get_pandas(self) -> pd.DataFrame:
...
 
 class ColumnMetric(Metric, Generic[ColumnTResult], abc.ABC):
-    column_name: str
+    column_name: ColumnName
+
+    def __init__(self, column_name: Union[ColumnName, str], options: AnyOptions = None):
+        self.column_name = ColumnName.from_any(column_name)
+        super().__init__(options)
+
+    # @property
```
Comment (Liraim) [inline src/evidently/base_metric.py:252]: Is this commented code needed here?

File: src/evidently/features/non_letter_character_percentage_feature.py [hunk 4]
```diff
@@ -7,8 +7,11 @@
 
 
 class NonLetterCharacterPercentage(GeneratedFeature):
+    column_name: str
+
     def __init__(self, column_name: str):
```
Comment (Liraim) [inline src/evidently/features/non_letter_character_percentage_feature.py:12]: I think `__init__` function can be removed.

File: src/evidently/features/text_length_feature.py [hunk 5]
```diff
@@ -7,8 +7,11 @@
 
 
 class TextLength(GeneratedFeature):
+    column_name: str
+
     def __init__(self, column_name: str):
```
Comment (Liraim) [inline src/evidently/features/text_length_feature.py:12]: Same, `__init__` function can be removed.

File: src/evidently/metrics/classification_performance/pr_table_metric.py [hunk 6]
```diff
@@ -22,12 +26,26 @@
...
+            return value
+
+
+PRTable = Dict[Union[LabelModel, Label], List[List[Union[float, int]]]]
 
 
 class ClassificationPRTableResults(MetricResult):
-    class Config:
-        smart_union = True
+    # class Config:
```
Comment (Liraim) [inline src/evidently/metrics/classification_performance/pr_table_metric.py:47]: Commented code.

File: src/evidently/metrics/data_drift/text_domain_classifier_drift_metric.py [hunk 7]
```diff
@@ -42,6 +42,7 @@ class TextDomainClassifierDriftMetric(Metric[TextDomainClassifierDriftResult]):
 
     def __init__(self, text_column_name: str) -> None:
```
Comment (Liraim) [inline src/evidently/metrics/data_drift/text_domain_classifier_drift_metric.py:43]: Same question about `__init__` with single parameter (can we delete this?).

File: src/evidently/metrics/data_quality/column_correlations_metric.py [hunk 8]
```diff
@@ -40,6 +40,7 @@ def __init__(self, column_name: Union[str, ColumnName]) -> None:
             self.column_name = column_name
         else:
             self.column_name = ColumnName.main_dataset(column_name)
```
Comment (Liraim) [inline src/evidently/metrics/data_quality/column_correlations_metric.py:42]: Can be replace by `ColumnName.from_any` as you introduced it.

File: src/evidently/metrics/regression_performance/error_bias_table.py [hunk 9]
```diff
@@ -48,13 +49,13 @@ class Config:
 
 class RegressionErrorBiasTable(Metric[RegressionErrorBiasTableResults]):
     # by default, we get 5% values for the error bias calculations
-    TOP_ERROR_DEFAULT = 0.05
-    TOP_ERROR_MIN = 0
-    TOP_ERROR_MAX = 0.5
+    TOP_ERROR_DEFAULT: ClassVar = 0.05
```
Comment (Liraim) [inline src/evidently/metrics/regression_performance/error_bias_table.py:52]: Probably better to set proper type, eg. `ClassVar[float]`

**Review by Liraim — CHANGES_REQUESTED** [review]

File: src/evidently/features/non_letter_character_percentage_feature.py (hunk 4 above)
Comment (mike0sv) [reply src/evidently/features/non_letter_character_percentage_feature.py:12]: It allows us to use positional args instead of kw-only

**Pull request merged**